### PR TITLE
Patch wc.data for Gutenberg 15.5+ and wc <= 7.6.0

### PR DESCRIPTION
**Pull request by chihsuan**
### Changes proposed in this Pull Request:

Closes # .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Remove Navigation disable code

- https://github.com/Automattic/wc-calypso-bridge/blob/f7cef0091e4c06d95cb009e60c17da4cf29cda3c/includes/class-wc-calypso-bridge-woocommerce-admin-features.php#L93-L95
- https://github.com/Automattic/wc-calypso-bridge/blob/f7cef0091e4c06d95cb009e60c17da4cf29cda3c/includes/class-wc-calypso-bridge-woocommerce-admin-features.php#L136-L137

4.Go to WooCommerce -> Settings -> Features
5. Enable the Navigation feature:
- Undo changes from this [PR](https://github.com/woocommerce/woocommerce/pull/36456/files) to enable the new navigation option
- Or run `wp option set woocommerce_navigation_enabled yes`
6. Confirm there are errors `peError: object null is not iterable (cannot read property Symbol(Symbol.iterator))` in dev console
7. Apply this patch
8. Confirm the no errors in the dev console
9. You might still not see the navigation menu in /wp-admin/admin.php?page=wc-settings&tab=advanced I'm not really sure why, but I think it's something related to bridge/atomic code
10. Go to wp-admin/admin.php?page=wc-admin 
11. Confirm navigation works

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: class-wc-calypso-bridge-shared.php [hunk 1]
```diff
@@ -133,6 +133,21 @@ public function add_scripts() {
 
 		wp_enqueue_script( 'wc-calypso-bridge' );
 		wp_enqueue_style( 'wc-calypso-bridge' );
+
+		// Inject the WC data store patch for WooCommerce <= 7.6 with Gutenberg 15.5+
+		$has_gutenberg             = is_plugin_active( 'gutenberg/gutenberg.php' );
+		$gutenberg_version         = $has_gutenberg ? get_plugin_data( WP_PLUGIN_DIR . '/gutenberg/gutenberg.php' )['Version'] : false;
+
+		if ( 
+			defined( 'WC_VERSION' ) && version_compare( WC_VERSION, '7.6.0', '<=' ) &&
```
Comment (adrianduffell) [inline class-wc-calypso-bridge-shared.php:142]:
```suggestion
			defined( 'WC_VERSION' ) && version_compare( WC_VERSION, '7.7.0', '<' ) &&
```

I think we'll make the Core fix only in WC 7.7 now. What do you think about changing it this?

Comment (chihsuan) [reply class-wc-calypso-bridge-shared.php:142]: Missed that. Sounds good, thanks!

**Review by rjchow — APPROVED** [review]
tis a work of art 🚀

**Comment by adrianduffell** [comment]
Tested great in 7.6, amazing work @chihsuan 🏅 🚀

**Pull request merged**